libsolidity/codegen/LValue.cpp

```diff
@@ -231,20 +231,14 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 				m_context
 					<< (u256(0x1) << (256 - 8 * dynamic_cast<FixedBytesType const&>(*m_dataType).numBytes()))
 					<< Instruction::SWAP1 << Instruction::DIV;
-			else if (
-				m_dataType->category() == Type::Category::Integer &&
-				dynamic_cast<IntegerType const&>(*m_dataType).isSigned()
-			)
+			else
 				// remove the higher order bits
 				m_context
 					<< (u256(1) << (8 * (32 - m_dataType->storageBytes())))
```

Comment:
I prefer an assertion `m_dataType->storageBytes() < 32`.  Currently, the storage bytes looks just not 32, either bigger or smaller.